code/modules/mob/living/simple_animal/hostile/mining/elites/herald.dm

```diff
@@ -0,0 +1,331 @@
+	universal_speak = TRUE
+	mouse_opacity = MOUSE_OPACITY_ICON
+	death_sound = 'sound/misc/demon_dies.ogg'
+	deathmessage = "begins to shudder as it becomes transparent..."
+	loot_drop = /obj/item/clothing/accessory/necklace/herald_cloak
+
+	can_talk = 1
+
+	attack_action_types = list(/datum/action/innate/elite_attack/herald_trishot,
+								/datum/action/innate/elite_attack/herald_directionalshot,
+								/datum/action/innate/elite_attack/herald_teleshot,
+								/datum/action/innate/elite_attack/herald_mirror)
+
+	var/mob/living/simple_animal/hostile/asteroid/elite/herald/mirror/my_mirror = null
+	var/is_mirror = FALSE
+
+/mob/living/simple_animal/hostile/asteroid/elite/herald/death()
+	. = ..()
+	if(!is_mirror)
+		addtimer(CALLBACK(src, .proc/become_ghost), 8)
+	if(my_mirror != null)
+		qdel(my_mirror)
+
+/mob/living/simple_animal/hostile/asteroid/elite/herald/proc/become_ghost()
+	icon_state = "herald_ghost"
+
+/mob/living/simple_animal/hostile/asteroid/elite/herald/say(message, bubble_type, list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null, filterproof = null)
+	. = ..()
+	playsound(get_turf(src), 'sound/magic/clockwork/invoke_general.ogg', 20, TRUE)
+
+/datum/action/innate/elite_attack/herald_trishot
+	name = "Triple Shot"
+	button_icon_state = "herald_trishot"
+	chosen_message = "<span class='boldwarning'>You are now firing three shots in your chosen direction.</span>"
+	chosen_attack_num = HERALD_TRISHOT
+
+/datum/action/innate/elite_attack/herald_directionalshot
+	name = "Circular Shot"
+	button_icon_state = "herald_directionalshot"
+	chosen_message = "<span class='boldwarning'>You are firing projectiles in all directions.</span>"
+	chosen_attack_num = HERALD_DIRECTIONALSHOT
+
+/datum/action/innate/elite_attack/herald_teleshot
+	name = "Teleport Shot"
+	button_icon_state = "herald_teleshot"
+	chosen_message = "<span class='boldwarning'>You will now fire a shot which teleports you where it lands.</span>"
+	chosen_attack_num = HERALD_TELESHOT
+
+/datum/action/innate/elite_attack/herald_mirror
+	name = "Summon Mirror"
+	button_icon_state = "herald_mirror"
+	chosen_message = "<span class='boldwarning'>You will spawn a mirror which duplicates your attacks.</span>"
+	chosen_attack_num = HERALD_MIRROR
+
+/mob/living/simple_animal/hostile/asteroid/elite/herald/OpenFire()
+	if(client)
+		switch(chosen_attack)
+			if(HERALD_TRISHOT)
+				herald_trishot(target)
+				if(my_mirror != null)
+					my_mirror.herald_trishot(target)
+			if(HERALD_DIRECTIONALSHOT)
+				herald_directionalshot()
+				if(my_mirror != null)
+					my_mirror.herald_directionalshot()
+			if(HERALD_TELESHOT)
+				herald_teleshot(target)
+				if(my_mirror != null)
+					my_mirror.herald_teleshot(target)
+			if(HERALD_MIRROR)
+				herald_mirror()
+		return
+	var/aiattack = rand(1,4)
+	switch(aiattack)
+		if(HERALD_TRISHOT)
+			herald_trishot(target)
+			if(my_mirror != null)
+				my_mirror.herald_trishot(target)
```

Comment:
```suggestion
				my_mirror?.herald_trishot(target)
```
I think this looks neater, cannot be bothered to do it for every single time this is done.

Reply:
Will do!